[SUPPORT] update balanceHistoryCache test

chabroA opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already.
Disclaimer: Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### 📝 Description

~Update some tests that seem deprecated (for a while).
Was raised by this PR https://github.com/LedgerHQ/ledger-live/pull/4886 but since it's not directly related to this PR, the update is done independently, since it also fails on develop as of today.
Tested on the [last commit](https://github.com/LedgerHQ/ledger-live/commit/a605eb96d6781c7406784c59581cbb604896e95f) that changed these files (cf. [this PR](https://github.com/LedgerHQ/ledger-live/pull/3521)), these tests where already failing back then.~

No changeset added since only updates tests (i.e: dev content, no actual logic update)

EDIT: 
The issue was actually not coming from the tests themselves but from the way the tests are run, and more specifically how timezones are handled.

The CI uses the package.json `test` script to run tests, which manually set the timezone to `TZ=America/New_York` through `cross-env TZ=America/New_York`. (PS: why use the `America/New_York` instead of the default `UTC` timezone? Good question).

But of you run the tests with a simple `pnpm jest` (which is a valid command and should work, it's just a matter of personal preferences), the timezone is not set and thus the system timezone is used, leading to tests failing if your system timezone is not `America/New_York`.

Since the tests should be consistent, regarding of how they are run, the timezone setting has been moved as part of the jest config.

Also added a test to ensure that the timezone used is indeed the `America/New_York` once to avoid future misunderstanding.

### ❓ Context

- https://github.com/LedgerHQ/ledger-live/pull/4886

- **Impacted projects**: `` <!-- Add the list of end user projects impacted by the change inside of the ` `, like so `LLM, LLD`. -->
- **Linked resource(s)**: [] <!-- Attach any ticket number if relevant inside the brackets, like so [LIVE-0000]. (JIRA / Github issue number) -->

### ✅ Checklist

- [x] **Test coverage** <!-- Are your changes covered by tests? Features must be tested, bugfixes must include a test that would have detected the issue. -->
- [x] **Atomic delivery** <!-- Is this pull request standalone? In order words, does it depend on nothing else? Please explain if not checked. -->
- [x] **No breaking changes** <!-- If there are breaking changes, please explain why. -->

### 📸 Demo

<!--
For visual features, please attach screenshots or video recordings to demonstrate the changes.
For libraries, you can add a code sample.
For bugfixes, you can drop this section.
-->

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

<!-- If any of the expectations are not met please explain the reason in detail. -->

gre commented:
do you know what broke this test before? are we still using nyc timezone for the test? i believe we're assessing somewhere around here that timezone behaves as expected.

chabroA commented:
> do you know what broke this test before? are we still using nyc timezone for the test? i believe we're assessing somewhere around here that timezone behaves as expected.

I updated the description and the commits. The issue was actually not coming from the tests themselves but from the way the tests are run, and more specifically how timezones are handled.

The CI uses the `package.json` `test` script to run tests, which manually set the timezone to TZ=America/New_York through cross-env TZ=America/New_York. (PS: why use the America/New_York instead of the default UTC timezone? Good question).

But if you run the tests with a simple `pnpm jest` (which is a valid command and should work, it's just a matter of personal preferences), the timezone is not set and thus the system timezone is used, leading to tests failing if your system timezone is not America/New_York.

Since the tests should be consistent, regarding of how they are run, the timezone setting has been moved as part of the jest config.

Also added a test to ensure that the timezone used is indeed the America/New_York once to avoid future misunderstanding.

gre commented:
> why use the America/New_York instead of the default UTC timezone? Good question

because when you use UTC, you may not reveal certain bug. for instance, the date to consider for a NYC time of 10PM is not the local YYYYMMDD date but the day after.

so if a logic is accidentally using the local date instead of the utc one, we can get into issue. this was specifically important for the caching of balance history where we had bad cache issue like this. for instance if you ask the backend what is the current rate for a future date (in the other case where you live in the future)
